ui-kit/src/main/kotlin/org/p2p/uikit/components/UiKitTip.kt

```diff
@@ -0,0 +1,164 @@
+package org.p2p.uikit.components
+
+import android.graphics.Point
+import android.view.LayoutInflater
+import android.view.View
+import android.widget.PopupWindow
+import androidx.annotation.StringRes
+import androidx.constraintlayout.widget.ConstraintLayout
+import androidx.constraintlayout.widget.ConstraintSet
+import androidx.core.view.updateLayoutParams
+import org.p2p.uikit.R
+import org.p2p.uikit.databinding.WidgetTipViewBinding
+import org.p2p.uikit.utils.toPx
+
+private enum class TipPosition {
+    TOP,
+    BOTTOM
+}
+
+/**
+ * Shows PopupWindow with arrow
+ * PopupWindow appears natively above or under the view
+ * Arrow is shown only on top or bottom of PopupWindow
+ * Arrow is pointing on the center of [anchorView]
+ */
+fun showTip(
+    anchorView: View,
+    counterText: String,
+    @StringRes title: Int,
+    @StringRes buttonNextTitle: Int = R.string.tip_button_next_title,
+    @StringRes buttonSkipTitle: Int = R.string.tip_button_skip_all_title,
+    onNextClick: () -> Unit,
+    onSkipClick: () -> Unit
+) {
+    val context = anchorView.context
+    showTip(
+        anchorView,
+        counterText,
+        context.getString(title),
+        context.getString(buttonNextTitle),
+        context.getString(buttonSkipTitle),
+        onNextClick,
+        onSkipClick
+    )
+}
+
+/**
+ * Shows PopupWindow with arrow
+ * PopupWindow appears natively above or under the view
+ * Arrow is shown only on top or bottom of PopupWindow
+ * Arrow is pointing on the center of [anchorView]
+ */
+fun showTip(
+    anchorView: View,
+    counterText: String,
+    title: String,
+    buttonNextTitle: String,
+    buttonSkipTitle: String,
+    onNextClick: () -> Unit,
+    onSkipClick: () -> Unit
+) {
+    val binding = WidgetTipViewBinding.inflate(LayoutInflater.from(anchorView.context)).apply {
+        countTextView.text = counterText
+        titleTextView.text = title
+        nextButton.text = buttonNextTitle
+        skipAllButton.text = buttonSkipTitle
+        nextButton.setOnClickListener { onNextClick() }
+        skipAllButton.setOnClickListener { onSkipClick() }
+    }.apply {
+        root.measure(View.MeasureSpec.UNSPECIFIED, View.MeasureSpec.UNSPECIFIED)
+    }
+
+    val popupView = binding.root
+
+    val popupWindow = PopupWindow(popupView, popupView.measuredWidth, popupView.measuredHeight, true).apply {
```

Comment:
i would do this constructor with separate line breaks for each param for Studio to show what each parameter means